packages/core/database/src/migrations/internal-migrations/5.0.0-02-document-id.ts

```diff
@@ -117,19 +117,11 @@ const migrateDocumentIdsWithLocalizations = async (db: Database, knex: Knex, met
 
 // Migrate document ids for tables that don't have localizations
 const migrationDocumentIds = async (db: Database, knex: Knex, meta: Meta) => {
-  let run = true;
+  const stream = knex.select('id').from(meta.tableName).whereNull('document_id').stream();
 
-  do {
-    const updatedRows = await knex(meta.tableName)
-      .update({ document_id: createId() })
-      .whereIn('id', (builder) => {
-        return builder.whereNull('document_id').select('id').limit(1);
-      });
-
-    if (updatedRows <= 0) {
-      run = false;
-    }
-  } while (run);
+  for await (const row of stream) {
```

Comment:
You can simply replace with a subquery that does a select from and not just this simple nested query, very straight forward and doesn't require any join

```
knex().update().whereIn('id', knex().select().from().whereNull().limit(1))
```

Reply:
I am doing this query and getting the same error:
```
    updatedRows = await knex(meta.tableName)
      .update({ document_id: createId() })
      .whereIn('id', knex(meta.tableName).select('id').whereNull('document_id').limit(1));
  } while (updatedRows > 0);
```

The query is translated to:
update `files` set `document_id` = 'ohvprwj9erke4gf328xl4n0c' where `id` in (select `id`   from `files` where `document_id` is null limit 1) 

And also returns the error:
This version of MySQL doesn't yet  support 'LIMIT & IN/ALL/ANY/SOME subquery

am I missing something?

Reply:
ho yes we do it in the codebase and have to actually do 

```
update ... 
where id IN SELECT id from (SELECT ....)) 
```

A double sub query, you can look at the runSubQuery code as we have it specifically for this usecase, maybe you could just use the queryBuilder actually

Reply:
I am getting the same error when executing this query:
```
update `tags`
set `document_id` = 'xs2b52m15oitv788qbvvtiz2'
where `id` in (
    select `id`
    from `tags`
    where
      `id` in (
        select `id`
        from `tags`
        where `document_id` is null
        limit 1
      )
  )
```

Looks like the issue is the "limit" clause,

Also, would it better to stream batch a find of ids and update them with a simple query for now, and spend time later  (once we get speed performance feedback) optimizing the query by batching the updates?

Reply:
the 1 sub query should look like that, don't use whereIn twice

select('id').from(knex(meta.tableName).select('id').whereNull('document_id').limit(1))

Reply:
About streams, it is very unclear what is really supported (what db does it natively or not etc). for pg at least I'm sure it's actually gonna be bad to use as we do not use the pg-query-stream lib and it just loads everything into memory 1st which is way worse.

Reply:
🫡 didn't know that about streams

Reply:
alright your suggestion is working ⭐  tested on mysql and sqlite

Reply:
I couldnt use the queryBuilder as I didnt find a way to reuse the existing transaction for it

Reply:
nice, I think it's automatically reusing the transaction btw